### ISSUE TITLE
Bors timeout when task fails (complete-push unscheduled vs. failed)

### ISSUE THREAD

**Issue by pocmo** [mozilla-mobile/android-components#4686]
We made bors wait for the `complete-push` task. However this task does not "fail fast". In the following task group one task failed and the `complete-push` task is just unscheduled due do that (instead of failed). So bors continues to wait until it times out. This slows down our merge queue since every failed task will cause bors to wait until its timeout.

https://tools.taskcluster.net/groups/PkeR4TLQS3eDHpOrBy6qfQ

**Comment by pocmo** [comment]
@JohanLorenzo Do you have an idea what we could do here? Is there a way how we could make this task fail?

**Comment by pocmo** [comment]
> This means all tasks will still have to run before the complete-push does. Therefore, it's a not a way to fail fast.

Not as fast as possible, but already significantly better. Is that a quick fix we could do easily?

> If we change taskcluster-github to prepend taskcluster/ to all Github Checks, then we can tell bors to watch for taskcluster/%. I'm not 100% sure, but it may enable fast failure.
Then, the complete-push task would become useless, and we could retire it.

That would be pretty neat! But also sounds like it will require more work / coordination.

**Comment by JohanLorenzo** [comment]
We'll be one step closer to getting emails once https://github.com/mozilla-mobile/reference-browser/pull/920 lands. I'll tackle the emails for builds after it lands.

**Comment by pocmo** [comment]
This has been quite annoying the last working days. There has been intermittent failures causing bors to do nothing for quite a while until the timeout is happening.

**Comment by JohanLorenzo** [comment]
Okay. I'll give a shot at the first solution tomorrow.

**Comment by JohanLorenzo** [comment]
Depends on https://phabricator.services.mozilla.com/D49402

**Comment by pocmo** [comment]
@JohanLorenzo Is this done? I think I still see this happen.

**Comment by JohanLorenzo** [comment]
Option 1 should be implemented. Do you have an example I can look into?

**Comment by pocmo** [comment]
@JohanLorenzo I do not remember but I'll make sure to post it here once I saw it.

I recently saw a slightly different variant that I remember though. In that case the decision task failed which caused no complete-push task to get scheduled at all. I wonder if it would help to add the decision task to bors.toml as an additional task it should look at? I hope that bors will fail as soon as one of the tasks fails and doesn't wait for all of them.
